src/MomentoCacheClient.php

```diff
@@ -2046,7 +2064,33 @@ public function zRemRangeByScore(string $key, string $start, string $end): Redis
      */
     public function zRevRange(string $key, int $start, int $end, mixed $scores = null): Redis|array|false
     {
-        throw MomentoToPhpRedisExceptionMapper::createCommandNotImplementedException(__FUNCTION__);
+        if ($start >= 0 && $end >= 0 && $start > $end) {
+            return [];
+        }
+        if ($start < 0 && $end < 0 && $start > $end) {
+            return [];
+        }
+
+        // Momento uses exclusive range for end rank, while PhpRedis uses inclusive range
+        $end = $end + 1;
```

Comment:
For the -1 case we need unbounded end, otherwise this will be 0. Let's add a test case for this situation.

Reply:
Doesn't this test case satisfy it:

```
 // Test valid rank range with positive start and negative end
        $elements = self::$client->zRevRange($key, 0, -1);
        $this->assertEquals([$member3, $member2, $member1], $elements, "Elements do not match expected order for mixed ranks");
```

Reply:
Yep that looks good